landoapi/landing_worker.py

```diff
@@ -69,6 +72,53 @@ def __init__(self, sleep_seconds=5):
         signal.signal(signal.SIGINT, self.exit_gracefully)
         signal.signal(signal.SIGTERM, self.exit_gracefully)
 
+    def _setup_ssh(self):
+        """Fetch a private SSH key from the environment and add it to ssh-agent.
+        """
+        SSH_PRIVATE_KEY_ENV_KEY = "SSH_PRIVATE_KEY"
+
+        # Fetch ssh private key from the environment. Note that this key should be
+        # stored in standard format including all new lines and new line at the end
+        # of the file.
+        ssh_private_key = os.environ.get(SSH_PRIVATE_KEY_ENV_KEY)
+        if not ssh_private_key:
+            logger.warning(f"No {SSH_PRIVATE_KEY_ENV_KEY} present in environment.")
+            return
+
+        # Set all the correct environment variables
+        agent_process = subprocess.run(
+            ["ssh-agent", "-s"], capture_output=True, universal_newlines=True
+        )
+
+        # This pattern will match keys and values, and ignore everything after the
+        # semicolon. For example, the output of `agent_process` is of the form:
+        #     SSH_AUTH_SOCK=/tmp/ssh-c850kLXXOS5e/agent.120801; export SSH_AUTH_SOCK;
+        #     SSH_AGENT_PID=120802; export SSH_AGENT_PID;
+        #     echo Agent pid 120802;
+
+        pattern = re.compile("(.+)=([^;]*)")
+        for key, value in pattern.findall(agent_process.stdout):
+            self.config["SSH_ENV_KEYS"].append(key)
+            os.environ[key] = value
```

Comment:
Nit: It seems like we really want to find the values of two specific variables here, `SSH_AUTH_SOCK` and `SSH_AGENT_PID`. Maybe we should be more explicit about finding the values of those keys instead of using a funky regex. This solution seems to work fine, so it's not a huge problem.

Reply:
I pondered this, but ended up using regex because the nature of `ssh-agent -s` is to generate both the keys and values that we need, and I figured this is future-proof, in case those keys change.

The regex also helps us extract those keys and values more easily. I could add explicit checks for those keys, but parsing the output manually without the regex is not necessarily cleaner. For example, we would have to handle the `echo` statement at the end, new lines, and doing splits to ignore text after the semicolon for the relevant lines.

For example, for the first two lines we'd have to do `key, value = line.split(";")[0].split("=")` which is not any cleaner than `re.compile("(.+)=([^;]*)")`. And we would have to do this only for lines 1 and 2 (i.e. we would have to either do `lines = agent_process.stdout.split("\n")[0:2]`, or check for presence of `=` or `;` on a line, neither which are pretty.)

Thoughts?

Reply:
Okay, I think you're right about future proofing and having this be cleaner that `split` + index + `split`, etc. I still think my concern about missing variables is warranted though. Could we log which variables this loop finds and adds to the environment? Then if we do hit that problem (`ssh-agent` changes or something is wonky with the regex) we can tell from the logs.

I also noticed that we create the `self.config["SSH_ENV_KEYS"]` list, but don't seem to do anything with it.

Reply:
Forgot to remove `self.config["SSH_ENV_KEYS"]` when I removed the tear down code, will do.

Agreed RE: env keys, will add proper logs to show which keys/values have been set.